st2api/st2api/controllers/v1/auth.py

```diff
@@ -97,11 +101,27 @@ def get_all(self, requester_user, show_secrets=None):
         """
         mask_secrets = self._get_mask_secrets(show_secrets=show_secrets,
                                               requester_user=requester_user)
-        api_key_dbs = ApiKey.get_all()
-        api_keys = [ApiKeyAPI.from_model(api_key_db, mask_secrets=mask_secrets)
-                    for api_key_db in api_key_dbs]
 
-        return api_keys
+        if limit and int(limit) > self.max_limit:
+            msg = 'Limit "%s" specified, maximum value is "%s"' % (limit, self.max_limit)
+            raise ValueError(msg)
+
+        api_key_dbs = ApiKey.get_all(limit=limit, offset=offset)
+
+        try:
+            api_keys = [ApiKeyAPI.from_model(api_key_db, mask_secrets=mask_secrets)
+                        for api_key_db in api_key_dbs]
+        except OverflowError:
```

Comment:
This is because of mongo limitation: ```2017-06-19 03:49:58,210 139885476219952 ERROR router [-] Failed to call controller function "get_all" for operation "st2api.co
ntrollers.v1.auth:api_key_controller.get_all": Python int too large to convert to C long
Traceback (most recent call last):
  File "/opt/stackstorm/st2/local/lib/python2.7/site-packages/st2common/router.py", line 414, in __call__
    resp = func(**kw)
  File "/opt/stackstorm/st2/local/lib/python2.7/site-packages/st2api/controllers/v1/auth.py", line 114, in get_all
    for api_key_db in api_key_dbs:
  File "/opt/stackstorm/st2/local/lib/python2.7/site-packages/mongoengine/queryset/queryset.py", line 97, in _iter_results
    self._populate_cache()
  File "/opt/stackstorm/st2/local/lib/python2.7/site-packages/mongoengine/queryset/queryset.py", line 116, in _populate_cache
    self._result_cache.append(self.next())
  File "/opt/stackstorm/st2/local/lib/python2.7/site-packages/mongoengine/queryset/base.py", line 1407, in next
    raw_doc = self._cursor.next()
  File "/opt/stackstorm/st2/local/lib/python2.7/site-packages/pymongo/cursor.py", line 1114, in next
    if len(self.__data) or self._refresh():
  File "/opt/stackstorm/st2/local/lib/python2.7/site-packages/pymongo/cursor.py", line 1036, in _refresh
    self.__collation))
  File "/opt/stackstorm/st2/local/lib/python2.7/site-packages/pymongo/cursor.py", line 873, in __send_message
    **kwargs)
  File "/opt/stackstorm/st2/local/lib/python2.7/site-packages/pymongo/mongo_client.py", line 905, in _send_message_with_respon
se
    exhaust)
  File "/opt/stackstorm/st2/local/lib/python2.7/site-packages/pymongo/mongo_client.py", line 916, in _reset_on_error
    return func(*args, **kwargs)
  File "/opt/stackstorm/st2/local/lib/python2.7/site-packages/pymongo/server.py", line 124, in send_message_with_response
    set_slave_okay, sock_info.is_mongos, use_find_cmd)
  File "/opt/stackstorm/st2/local/lib/python2.7/site-packages/pymongo/message.py", line 291, in get_message
    spec, self.fields, self.codec_options)
OverflowError: Python int too large to convert to C long```. 
For these endpoints we don't have pagination handling in the base controller. No, overflow error is only thrown when the offset > 8 byte int